docs/src/docs/asciidoc/user/serializers-copiers.adoc

```diff
@@ -125,40 +125,37 @@ of the serialized types as they might not be available in the current class load
-In the case of a disk persistent cache, the contents of the map will be persisted locally on to the disk.
-For clustered caches the contents are persisted in the cluster itself so that other clients using the same cache can also access the contents of the map.
-
-Attempting to configure a serializer that lacks such constructor on a persistent cache using either of
+Attempting to configure a serializer that lacks such constructor on a cache using either of
 `CacheConfigurationBuilder.withKeySerializer(Class<? extends Serializer<K>> keySerializerClass)` or
 `CacheConfigurationBuilder.withValueSerializer(Class<? extends Serializer<V>> valueSerializerClass)`
 will be sanctioned with an exception upon cache initialization.
 
-Configuring a serializer that lacks such constructor by instance on a persistent cache using either of
-`CacheConfigurationBuilder.withKeySerializer(Serializer keySerializer)` or `CacheConfigurationBuilder.withValueSerializer(Serializer valueSerializer)`
-will work, but the responsibility of persisting and restoring the serializer's state across restarts lies on you.
+But if an instance of the serializer is configured using either of
+`CacheConfigurationBuilder.withKeySerializer(Serializer keySerializer)` or
+`CacheConfigurationBuilder.withValueSerializer(Serializer valueSerializer)`
+it will work since the instantiation is done by the user code itself.
 
-On caches that have no persistent capable store, serializers must have a constructor:
+Registering a serializer that lacks such constructor at the cache manager level will prevent it from being chosen for caches.
+
+Custom serializer implementations could have some state that is used in the serialization/deserialization process.
+When configured on a persistent cache, such serializers may need to persist and restore their state across restarts.
+For clustered caches there might be an additional requirement that the state of the serializer must be visible to all clients using the same cache(configured with the same serializer).
+To address these requirements you can have a `StatefulSerializer` implementation.
+`StatefulSerializer` is a specialized `Serializer` with an additional _init_ method with the following signature:
 ```java
-public MySerializer(ClassLoader classLoader) {
+public init(StateRepository repository) {
 }
 ```
-Attempting to configure a serializer that lacks such constructor on a transient cache using either of
-`CacheConfigurationBuilder.withKeySerializer(Class<? extends Serializer<K>> keySerializerClass)` or
-`CacheConfigurationBuilder.withValueSerializer(Class<? extends Serializer<V>> valueSerializerClass)`
-will be sanctioned with an exception upon cache initialization.
-
-Registering a serializer that lacks such constructor at the cache manager level will prevent it from being chosen for persistent caches.
+The `StateRepository.getPersistentConcurrentMap()` provides a `ConcurrentMap` that you can use to store any relevant state.
```

Comment:
Add `String, Class<K>, Class<V>` to indicate that `getPersistentConcurrentMap` takes parameters.